### PR TITLE
PR: Restructure console menus

### LINKED ISSUE #8478: Improve IPython console Options menu
The Options menu of the IPython console has grown a lot lately:

seleccion_001

My idea is to shrink it a bit, like this:

```
New console
Specialized consoles > Pylab / Sympy / Cython
----
Restart kernel
Connect to an existing kernel
Remove all variables
----
Rename tab
Show env vars
Show sys.path contents
Show elapsed time
```

Notes:
* `Specialized consoles` would be a submenu but only of the IPython console options menu. The main `Consoles` menu wouldn't have it.
* The last two actions in the screenshot above will be added to all plugins after PR #8192.

@spyder-ide/core-developers, what do you think?

### PR DESCRIPTION
**Pull request by juanis2112**
## Description of Changes

* [ ] Wrote at least one-line docstrings (for any new functions)
* [ ] Added unit test(s) covering the changes (if testable)
* [x] Included a screenshot (if affecting the UI)

Rearranged IPython console options menu and IPython main console options menu. 

cli

top

### Issue(s) Resolved

Fixes #8478 

### Affirmation

By submitting this Pull Request or typing my (user)name below,
I affirm the [Developer Certificate of Origin](https://developercertificate.org)
with respect to all commits and content included in this PR,
and understand I am releasing the same under Spyder's MIT (Expat) license.

I certify the above statement is true and correct: Juanis2112

### CODE REVIEW

**Comment by ccordoba12** [comment]
@juanis2112, to do the rebase you need to run these commands

```
git checkout master
git fetch upstream
git merge upstream/master
git checkout Solve_issue8478
git rebase master
```

**Comment by CAM-Gerlach** [comment]
> git merge upstream/master

Is there a reason to use this over ``git pull upstream master``? I'm just curious if I've been doing it wrong all these years, lol.

**Comment by CAM-Gerlach** [comment]
See my comment on #8478 .

File: spyder/plugins/ipythonconsole/widgets/client.py [hunk 1]
```diff
@@ -379,13 +379,17 @@ def get_kernel(self):
...
+                                     triggered=self.stop_button_click_handler)
+               
         reset_action = create_action(self, _("Remove all variables"),
                                      icon=ima.icon('editdelete'),
                                      triggered=self.reset_namespace)
 
         env_action = create_action(
                         self,
-                        _("Show environment variables"),
+                        _("Show env vars"),
```
Comment (CAM-Gerlach) [inline spyder/plugins/ipythonconsole/widgets/client.py]: ``"Show environment variables"`` is clearer to those who may not know the meaning of the abbreviations (non-natives, newer users, etc) and still shorter than ``Connect to an existing kernel`` so it doesn't make the menu any wider anyway.

**Comment by ccordoba12** [comment]
@juanis2112, there are still some pep8 issues to solve.

**Comment by CAM-Gerlach** [comment]
Also, once its final, @juanis2112 should update the screenshot in the OP.

**Comment by dalthviz** [comment]
@ccordoba12 @CAM-Gerlach I have updated the screenshot of the menu, what do you think?

**Comment by ccordoba12** [comment]
> I have updated the screenshot of the menu, what do you think?

cli

This menu is done for me, so please don't touch it any longer.

> top

We agreed with @CAM-Gerlach to also have here the `New special console` submenu below `New console`, so that's still missing. @juanis2112, please address that here.

----

By the way, great work @juanis2112 and @dalthviz!! This looks really good!

**Comment by juanis2112** [comment]
> > I have updated the screenshot of the menu, what do you think?
> 
> cli
> 
> This menu is done for me, so please don't touch it any longer.
> 
> top
> 
> We agreed with @CAM-Gerlach to also have here the `New special console` submenu below `New console`, so that's still missing. @juanis2112, please address that here.
> 
> By the way, great work @juanis2112 and @dalthviz!! This looks really good!

Thanks @ccordoba12. I just added the special console menu in the main consoles menu.

screen shot 2019-01-30 at 12 14 01 pm

**Comment by ccordoba12** [comment]
Thanks @juanis2112!

**Comment by ccordoba12** [comment]
> However, it seems like the consensus is firmly against adding them

Well, we can always revise our decision. @bcolsen, @jnsebgosselin, @jitseniesen, do you have something to say about this?

**Comment by jitseniesen** [comment]
I am also against adding the items. Environment variables seems very specialized. The path less so but it still feels out of place to me.

**Comment by ccordoba12** [comment]
Ok, I think we reached a consensus. Thanks @jitseniesen for the input.

**Comment by dalthviz** [comment]
@juanis2112 could you fix the conflicts to merge this? Thanks :)

**Comment by ccordoba12** [comment]
@dalthviz, please merge if this is ready.

**Pull request merged**